### PR TITLE
Add grafana documentation

### LINKED ISSUE #92: Documentation for instructions on adding new grafana dashboards
Based on our discussion in #87, we should add documentation for adding new grafana dashboards
(also related to https://github.com/operate-first/SRE/issues/5)

### PR DESCRIPTION
**Pull request by hemajv**
Including documentation on how to add grafana dashboards

This closes #92.

### CODE REVIEW

File: docs/odh/grafana/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Grafana Monitoring Dashboards
+
+[Grafana](https://grafana.com/oss/grafana) is a multi-platform open source analytics and interactive visualization tool. It allows you to query, create dashboards, explore, and alert on metrics when connected to supported data sources.
+
+For each of the Operate First managed services and applications, we have created example Grafana dashboards which can be used to visualize our [SLI/SLOs](https://github.com/operate-first/SRE/tree/master/sli-slo) defined for each of the applications.
```
Comment (HumairAK) [inline docs/odh/grafana/README.md:5]: I would rephrase this to say: 
"For various Operate First managed services and applications..."

The distinction being, that in the original version, we're saying we already have dashboards for _all_ the operate-first managed services, which is not the case. Hope that makes sense.

File: docs/odh/grafana/add_grafana_dashboard.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Add a grafana dashboard to OperateFirst Grafana
+
+To be able to include your dashboards in Grafana, you will need to add a `GrafanaDashboard` custom resource. This custom resource is defined as a yaml file that downloads the dashboard contents from a specified URL and falls back to embedded json if the URL can not be resolved.
```
Comment (HumairAK) [inline docs/odh/grafana/add_grafana_dashboard.md:3]: Replace: 
> This custom resource is defined as a yaml file that downloads the dashboard contents 

with: 
> This custom resource is defined as a yaml file that can remotely reference the dashboard contents

File: docs/odh/grafana/add_grafana_dashboard.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  name: grafana-dashboard-from-url
+  labels:
+    app: grafana
+spec:
+  name: grafana-dashboard-from-url.json
+  url: https://raw.githubusercontent.com/integr8ly/grafana-operator/master/deploy/examples/remote/grafana-dashboard.json
+  json:
+```
+
+In this template, all you need to do is update the dashboard name, the dashboard JSON file name and the URL pointing to the location of your dashboard JSON file.
```
Comment (HumairAK) [inline docs/odh/grafana/add_grafana_dashboard.md:21]: I would be a bit verbose here, and  also list the actual yaml paths, e.g. instead of just "dashboard name" also say "spec.name".

File: docs/odh/grafana/add_grafana_dashboard.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+Pick a suitable name, ensure that its unique in the [`odh/base/monitoring/overrides/grafana-operator/overlays/dashboards`](https://github.com/operate-first/apps/tree/master/odh/base/monitoring/overrides/grafana-operator/overlays/dashboards) folder.
```
Comment (HumairAK) [inline docs/odh/grafana/add_grafana_dashboard.md:23]: Suitable name, as in `metadata.name` ? let's specify that as well.

**Comment by HumairAK** [comment]
Other than the above comments, this lgtm.

File: docs/odh/grafana/README.md (hunk 1 above)
Comment (hemajv) [reply docs/odh/grafana/README.md]: yeah that makes sense, updated it :+1:

File: docs/odh/grafana/add_grafana_dashboard.md (hunk 2 above)
Comment (hemajv) [reply docs/odh/grafana/add_grafana_dashboard.md]: fixed :+1:

File: docs/odh/grafana/add_grafana_dashboard.md (hunk 3 above)
Comment (hemajv) [reply docs/odh/grafana/add_grafana_dashboard.md]: yes I was actually thinking the same too. modified it

File: docs/odh/grafana/add_grafana_dashboard.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDashboard
+metadata:
+  name: grafana-dashboard-from-url
+  labels:
+    app: grafana
+spec:
+  name: grafana-dashboard-from-url.json
+  url: https://raw.githubusercontent.com/integr8ly/grafana-operator/master/deploy/examples/remote/grafana-dashboard.json
+  json:
```
Comment (tumido) [inline docs/odh/grafana/add_grafana_dashboard.md:18]: If both the `url` and `json` properties are specified, the `json` value is used only if the `url` is unresolvable (it's a fallback value). I think it's not necessary to keep it here, I would rather see the operator fail to load the dashboard (so we can track it and alert on it) than fallback to an "empty" one, wdyt?

reference:
- https://github.com/integr8ly/grafana-operator/blob/master/documentation/dashboards.md
- https://github.com/integr8ly/grafana-operator/issues/45#issuecomment-529467580

File: docs/odh/grafana/add_grafana_dashboard.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+In this template, all you need to do is update:
+1. `metadata.name` - the dashboard name
+2. `spec.name` - the dashboard JSON file name
+3. `spec.url` - the URL pointing to the location of your dashboard JSON file
+
+***Note:** Ensure that the `spec.json` field is empty if setting the URL, if both url and json are specified then the json field won't be updated when fetched.*
```
Comment (tumido) [inline docs/odh/grafana/add_grafana_dashboard.md:26]: See above. I would avoid specifying both `url` and `json` fields at the same time. Let's either inline it (`json`) or have an `url` without a "magic" empty `json` field fallback.

**Review by tumido — COMMENTED** [review]
I would change the `json`/`url` fields handling in the `GrafanaDashboard` a bit. WDYT?

File: docs/odh/grafana/add_grafana_dashboard.md (hunk 5 above)
Comment (hemajv) [reply docs/odh/grafana/add_grafana_dashboard.md:18]: @tumido good poin! I think it would be nice if we can alert on it as well, will retain only the `url` field then

**Review by tumido — APPROVED** [review]
LGTM! :+1: Nice piece of documentation! :rocket:

**Review by HumairAK — APPROVED** [review]
